RELEASE/scripts/autoscend/auto_restore.ash

```diff
@@ -0,0 +1,1261 @@
+ *
+ * returns true if my_mp() >= goalPercent after attempting to restore.
+ */
+boolean acquireMP(float goalPercent){
+	return acquireMP(goalPercent, can_interact());
+}
+
+/**
+ * Try to acquire up to the mp goal expressed as a percentage, optionally buying items (useFreeRests: true). Will also cure poisoned and beaten up before restoring any mp.
+ *
+ * returns true if my_mp() >= goalPercent after attempting to restore.
+ */
+boolean acquireMP(float goalPercent, boolean buyItems){
+	return acquireMP(goalPercent, buyItems, true);
+}
+
+/**
+ * Try to acquire up to the mp goal expressed as a percentage (out of either 1.0 or 100.0), optionally buying items and using free rests. Will also cure poisoned and beaten up before restoring any mp.
+ *
+ * returns true if my_mp() >= goalPercent after attempting to restore.
+ */
+boolean acquireMP(float goalPercent, boolean buyItems, boolean useFreeRests){
+	int goal = my_maxmp();
+	if(goalPercent > 1.0){
+		goal = ceil((goalPercent/100.0) * my_maxmp());
+	} else{
+		goal = ceil(goalPercent*my_maxmp());
+	}
+	return acquireMP(goal.to_int(), buyItems, useFreeRests);
+}
+
+/**
+ * Try to acquire your max hp (buyItems: false, useFreeRests: true). Will also cure poisoned and beaten up before restoring any hp.
+ *
+ * returns true if my_hp() >= my_maxhp() after attempting to restore.
+ */
+boolean acquireHP(){
+	return acquireHP(my_maxhp());
+}
+
+/**
+ * Try to acquire up to the hp goal (buyItems: false, useFreeRests: true). Will also cure poisoned and beaten up before restoring any hp.
+ *
+ * returns true if my_hp() >= goal after attempting to restore.
+ */
+boolean acquireHP(int goal){
+	return acquireHP(goal, can_interact());
+}
+
+/**
+ * Try to acquire up to the hp goal, optionally buying items (useFreeRests: true). Will also cure poisoned and beaten up before restoring any hp.
+ *
+ * returns true if my_hp() >= goal after attempting to restore.
+ */
+boolean acquireHP(int goal, boolean buyItems){
+	return acquireHP(goal, buyItems, true);
+}
+
+ /**
+  * Try to acquire up to the hp goal, optionally buying items and using free rests. Will also cure poisoned and beaten up before restoring any hp.
+  *
+  * returns true if my_hp() >= goal after attempting to restore.
+  */
+boolean acquireHP(int goal, boolean buyItems, boolean useFreeRests){
+
+  __cure_bad_stuff();
+
+  if(goal > my_maxhp()){
+		return false;
+	}
+
+	if(my_hp() >= goal){
+		return true;
+	}
+
+  __restore("hp", goal, buyItems, useFreeRests);
+
+	return my_hp() >= goal;
+}
+
+/**
+ * Try to acquire up to the hp goal expressed as a percentage (out of either 1.0 or 100.0) (buyItems: false, useFreeRests: true). Will also cure poisoned and beaten up before restoring any hp.
+ *
+ * returns true if my_hp() >= goalPercent after attempting to restore.
+ */
+boolean acquireHP(float goalPercent){
```

Comment:
Nit: I think this should be a different name, just in case someone accidentally calls this with something like `acquireHP(my_hp() / 2.0)`